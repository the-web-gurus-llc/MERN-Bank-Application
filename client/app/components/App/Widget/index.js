import React from 'react';
import PropTypes from 'prop-types';
import LoadingIndicator from 'components/LoadingIndicator';
import {
  StyledWidget,
  StyledWidgetContent,
  StyledWidgetContentWidget,
  StyledWidgetContentTypography,
  StyledWidgetContentTypographyHeader,
  StyledWidgetContentTypographyMain,
  StyledWidgetContentTypographyMainUnit,
  StyledWidgetContentDescription,
} from './styles';

export default function Widget({
  isLoading,
  svg,
  svgFormat,
  title,
  content,
  unit,
  description,
}) {
  return (
    <StyledWidget>
      {isLoading ? (
        <LoadingIndicator />
      ) : (
        <StyledWidgetContent centered={!!description}>
          {title && content && unit && (
            <StyledWidgetContentTypography>
              <StyledWidgetContentTypographyHeader>
                {title}
              </StyledWidgetContentTypographyHeader>

              <StyledWidgetContentTypographyMain>
                {content}
                <StyledWidgetContentTypographyMainUnit>
                  {unit}
                </StyledWidgetContentTypographyMainUnit>
              </StyledWidgetContentTypographyMain>
            </StyledWidgetContentTypography>
          )}

          {description && (
            <StyledWidgetContentDescription>
              {description}
            </StyledWidgetContentDescription>
          )}

          {svg && (
            <StyledWidgetContentWidget svgFormat={svgFormat}>
              {svg}
            </StyledWidgetContentWidget>
          )}
        </StyledWidgetContent>
      )}
    </StyledWidget>
  );
}

Widget.propTypes = {
  isLoading: PropTypes.bool,
  svg: PropTypes.object,
  svgFormat: PropTypes.bool,
  title: PropTypes.string,
  content: PropTypes.string,
  unit: PropTypes.string,
  description: PropTypes.string,
};
