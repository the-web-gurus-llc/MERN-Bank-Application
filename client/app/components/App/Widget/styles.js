import styled from 'styled-components';
import { typography } from 'utils';

export const StyledWidget = styled.section`
  height: 96px;
  background-color: rgb(243, 243, 243);
  border: 1px solid rgba(0, 0, 0, 0.12);
  padding: 16px 24px;
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
`;

export const StyledWidgetContent = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: ${({ centered }) => (centered ? 'baseline' : 'center')};
  height: 100%;
  width: 100%;
`;

export const StyledWidgetContentTypography = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  line-height: 1.4;
  width: 100%;
`;

export const StyledWidgetContentTypographyHeader = styled.div`
  font-size: 18px;
`;

export const StyledWidgetContentTypographyMain = styled.div`
  font-size: 25px;
  font-weight: ${typography.fontWeightMedium};
`;

export const StyledWidgetContentTypographyMainUnit = styled.span`
  font-size: 16px;
  margin-left: 5px;
  font-weight: ${typography.fontWeightLight};
`;

export const StyledWidgetContentDescription = styled.div`
  font-size: 14px;
`;

export const StyledWidgetContentWidget = styled.div`
  width: 100%;
  max-width: 74px;
  display: flex;

  .recharts-wrapper {
    width: 100% !important;
    position: relative;
  }

  svg {
    width: 100%;
    height: 100%;

    &.recharts-surface {
      max-width: 74px;
      position: absolute;
      right: -6px;
    }

    max-width: 74px;
  }
`;
