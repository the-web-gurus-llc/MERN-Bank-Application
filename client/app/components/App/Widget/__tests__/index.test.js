import React from 'react';
import { render } from '@testing-library/react';
import Widget from '../index';

describe('<Widget />', () => {
  it('should render a Widget', () => {
    const { container } = render(<Widget />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
