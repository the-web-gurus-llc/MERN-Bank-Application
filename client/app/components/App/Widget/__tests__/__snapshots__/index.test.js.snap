// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Widget /> should render a Widget 1`] = `
<section
  class="Widgetstyle__StyledWidget-sc-1bl1ktb-0 kjDbLh"
>
  <div
    class="Widgetstyle__StyledWidgetContent-sc-1bl1ktb-1 gBcXHr"
  />
</section>
`;
